lib/types/src/serialize.rs

```diff
@@ -141,25 +119,16 @@ impl SerializableModule {
     pub fn archive_from_slice_checked(
         metadata_slice: &[u8],
     ) -> Result<&ArchivedSerializableModule, DeserializeError> {
-        if metadata_slice.len() < 8 {
-            return Err(DeserializeError::Incompatible(
-                "invalid serialized data".into(),
-            ));
-        }
-        let mut pos: [u8; 8] = Default::default();
-        pos.copy_from_slice(&metadata_slice[metadata_slice.len() - 8..metadata_slice.len()]);
-        let pos: u64 = u64::from_le_bytes(pos);
-        check_archived_value::<Self>(&metadata_slice[..metadata_slice.len() - 8], pos as usize)
-            .map_err(|err| DeserializeError::CorruptedBinary(err.to_string()))
+        rkyv::access::<_, rkyv::rancor::Error>(metadata_slice)
+            .map_err(|e| DeserializeError::CorruptedBinary(e.to_string()))
     }
 
     /// Deserialize a compilation module from an archive
     pub fn deserialize_from_archive(
         archived: &ArchivedSerializableModule,
     ) -> Result<Self, DeserializeError> {
-        let mut deserializer = SharedDeserializeMap::new();
-        RkyvDeserialize::deserialize(archived, &mut deserializer)
-            .map_err(|e| DeserializeError::CorruptedBinary(format!("{:?}", e)))
+        rkyv::deserialize::<_, rkyv::rancor::Error>(archived)
+            .map_err(|e| DeserializeError::CorruptedBinary(e.to_string()))
     }
 
     /// Create a `ModuleInfo` for instantiation
```

Comment:
Must bump the CURRENT_VERSION in this file to `8`.

Reply:
[0e6219c](https://github.com/wasmerio/wasmer/pull/5142/commits/0e6219c6b305145ec91110b1d125e30e51f062d8)